src/test/java/com/hubspot/jinjava/lib/tag/IncludeTagTest.java

```diff
@@ -188,4 +188,16 @@ public void itSetsErrorLineNumbersCorrectlyTwoLevelsDeep() throws IOException {
     assertThat(result.getErrors().get(0).getSourceTemplate().get())
       .isEqualTo("tags/includetag/errors/error.html");
   }
+
+  @Test
+  public void itRunsWithoutTagCycleException() throws Exception {
+    String result = jinjava.render(
+      Resources.toString(
+        Resources.getResource("tags/extendstag/tagcycleexception/main-template.jinja"),
```

Comment:
I find it a bit easier to reason about these import chains if you rename them to something sequential like template-a template-b,template-c so  you can easily track a extends b imports c etc.

Reply:
Ok, templates were renamed. To better understanding please check the inheritance tree:

![image](https://user-images.githubusercontent.com/29970367/84246786-4e157e00-ab31-11ea-8d7c-e70361d24b5b.png)